test/clients/azure/osa.go

```diff
@@ -24,3 +33,28 @@ func (cli *Client) OSAResourceGroup(ctx context.Context, resourcegroup, name, lo
 	rgid := *app.ApplicationProperties.ManagedResourceGroupID
 	return rgid[strings.LastIndexByte(rgid, '/')+1:], nil
 }
+
+// UpdateOSACluster updates an OpenshiftManagedCluster by providing both the
+// currently applied external manifest and an updated internal manifest
+func (cli *Client) UpdateOSACluster(ctx context.Context, external *v20180930preview.OpenShiftManagedCluster, config *api.PluginConfig) (*v20180930preview.OpenShiftManagedCluster, error) {
+	// Remove the provisioning state before updating, if set
+	external.Properties.ProvisioningState = nil
+
+	// create logger
+	logrus.SetLevel(log.SanitizeLogLevel("Debug"))
+	logrus.SetFormatter(&logrus.TextFormatter{FullTimestamp: true})
+	logger := logrus.NewEntry(logrus.StandardLogger())
+	var oc *v20180930preview.OpenShiftManagedCluster
+	var err error
+
+	// simulate the API call to the RP
+	if err := wait.PollImmediate(5*time.Second, 1*time.Hour, func() (bool, error) {
```

Comment:
You don't need PollImmediate ever since you removed the error handling.

Reply:
this is not even needed for scaleupdown since scaleupdown should speak to the RP through the client library.